back-end/index.js

```diff
@@ -164,11 +164,17 @@ app.patch('/profile/:username/picture', async (req, res) => {
     }
 });
 
-const recommendations = async (courses, level, rating) => {
+const recommendations = async (courses, level, rating, currCourse, username) => {
+    filteredCourses = []
     //filtering out the courses that don't have the same difficulty and courses thats rating is below what the user rates themselves
-    const filteredCourses = courses.filter(course => {
-        return course.difficulty === level && course.avgRating >= rating;
-    });
+    for(let i = 0; i < courses.length; i++){
+        let curr = courses[i]
+        for (let j = 0; j < curr.completedBy.length; i++){
+            if(curr.difficulty === level && curr.avgRating >= rating && curr.title != currCourse && curr.completedBy[j] != username){
```

Comment:
What is the purpose of changing filter to a nested for? They're doing the same thing